### ISSUE TITLE
How this can be used with Stackdriver Error Reporting?

### ISSUE THREAD

**Issue by kop** [GoogleCloudPlatform/fluent-plugin-google-cloud#99]
I'm a little bit confused about how to report errors to Stackdriver Error Reporting by sending JSON strings to stdout. And is it even possible?

I'm trying to use JSON format described in https://cloud.google.com/error-reporting/docs/formatting-error-messages , log messages appear in Stackdriver Logging, but they never appear in Error Reporting.

**Comment by munangst** [comment]
@thomasschickinger can you answer this?

**Comment by igorpeshansky** [comment]
Actually, I can. @kop, the format you quoted is for sending errors directly to the error reporting API via REST calls. If you want to use the logging integration with error reporting, you should send regular log payloads with the error traces in the "log" or "message" field. The error reporting integration will pick it up from the ingested logs in Stackdriver Logging backend.
If you write JSON to stdout, I assume you're in some sort of container environment. How is your stdout sent to fluentd? You might need to change the configuration to indicate `format json` to the input plugin, so the payload gets parsed appropriately (we do have JSON detection, but it may not always work).
Alternatively, with the default `google-fluentd` package you can just write your JSON payload directly to port 24224 on your machine.
Hope this helps.

**Comment by kop** [comment]
Thank you, @igorpeshansky.

I'm using GKE, and after digging in the source code of K8S and some other places, came here because had a feel that this is the right place to ask this question. :)

I will try `message` field, however, as I understand, it expects error trace. But how can I add more data, like [ServiceContext](https://cloud.google.com/error-reporting/reference/rest/v1beta1/ServiceContext) and [ErrorContext](https://cloud.google.com/error-reporting/reference/rest/v1beta1/ErrorContext)?

**Comment by thomasschickinger** [comment]
Hi @kop, the metadata in the Error Reporting JSON format (like ServiceContext) is also taken into account when sending errors through logs and not via the API. The important part is that your log is handled as JSON log and not as text log. You can check that by looking at the payload field in Stackdriver Logging. It should be called jsonPayload and not textPayload.

**Comment by kop** [comment]
Thank you, everyone. I got it working.

For those who might come here from Google, this works just fine:

```js
console.log(
  JSON.stringify({
    message: error.stack,
    severity: 'ERROR',
    serviceContext: {
      service: 'my-service',
      version: '1.0.0'
    },
    context: {
      httpRequest: {
        method: 'GET',
        url: 'https://mysite.com/foo/bar',
        responseStatusCode: 500,
        remoteIp: '127.0.0.1'
      }
    }
  })
);
```

**Comment by swizzlr** [comment]
@kop @igorpeshansky weirdly, this used to work for me and stopped last week when I upgraded my cluster to run new nodes. any ideas?

**Comment by igorpeshansky** [comment]
There's been some breakage in GKE JSON parsing in version 1.6.6. It'll be fixed in 1.6.7.

**Comment by swizzlr** [comment]
@igorpeshansky WHEW. That'll teach me to sit on the cutting edge!
